### ISSUE TITLE
Trend Micro Apex One cannot ingest some kinds of CEF log into Sentinel

### ISSUE THREAD

**Issue by vincentche1** [Azure/Azure-Sentinel#10455]
**Describe the bug**
Some kinds of Trend Micro Apex One CEF logs are not ingested into CommonSecurityLog Table

**To Reproduce**
1. Query in Syslog table:
Syslog
| where SyslogMessage contains 'Trend Micro'
| where SyslogMessage contains 'DETECTION'

Result:
2024-05-08T16:07:11Z forwarder.tmes.trendmicro.com tmes[1]: CEF:0|Trend Micro|TMES|1.0.0.0|100101|DETECTION|6|rt=2024-05-08T16:06:51.552Z cs1Label=eventType cs1=anti_spoof cs2Label=domainName cs2=pinelabs.com suser= duser=cards.msg4-nr.ib@pinelabs.com cs3Label=direction cs3=incoming cs4Label=messageId cs4=<4678d2$2la1jo@esa8.hc2819-66.iphmx.com> msg=Delivery Status Notification (Failure) cn1Label=messageSize cn1=12909 cs5Label=policyName cs5=DMARC Rule act=Quarantine cs6Label=details cs6={}

2, Query in CommonSecurityLog table
CommonSecurityLog
| where DeviceVendor == 'Trend Micro'
| where Activity == 'DETECTION'

No result

**Expected behavior**
If I execute the query in CommonSecurityLog, there will be CEF log of Trend Micro Apex One.
I checked the CEF log format and it is eligible but why it isn't ingested into CommonSecurityLog table?
https://docs.trendmicro.com/en-us/documentation/article/trend-micro-email-security-online-help-syslog-content-mappi

**Comment by v-sudkharat** [comment]
Hi @vincentche1, Thanks for flagging this issue, we will investigate this issue and get back to you with some updates by 16-05-2024. Thanks!

**Comment by vincentche1** [comment]
Dear @v-sudkharat, I hope you can expedite faster as the problem is impacted customer workflow. Please inform me if you need further details for investigation.
The parser of Trend Micro Apex One CEF logs link: https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/Trend%20Micro%20Apex%20One/Parsers/TMApexOneEvent.yaml

**Comment by v-sudkharat** [comment]
@vincentche1, Sure, and Thanks for providing details.

**Comment by vincentche1** [comment]
Hi @v-sudkharat, we hope that you can expedite this issue as it is pending for a remarkable time

**Comment by v-sudkharat** [comment]
Hi @vincentche1, we were unable to address this issue because of our lack of availability. we'll investigate on it and give you an update by 25-05-2024. Thanks!

**Comment by v-sudkharat** [comment]
Hi @vincentche1, Just want to check below few points with your end: - 
1. As per your give description, Your facing issue in step C. where data is not received into the CommonSecurityLog table, Is that correct? 
![image](https://github.com/Azure/Azure-Sentinel/assets/132428394/a6e18ae5-3785-4c58-b68e-1039ee83d4e3)


2. Could you please check and let us know what is "streams" name for existing created Data Collection Rule? Please share the JSON view with us if possible: -
![image](https://github.com/Azure/Azure-Sentinel/assets/132428394/c975ceb9-b3d4-4ec5-962b-5bd2e412756b)

2. Could you please validate and let us know the below configuration has been done in Azure and Apex portal which mentioned in steps A and B: - 
![image](https://github.com/Azure/Azure-Sentinel/assets/132428394/69226ebf-1973-4a55-97fe-8da43ecac3de)

Thanks!

**Comment by v-sudkharat** [comment]
@vincentche1, Ok. Noted. Let us check and will update you.

**Comment by MeirLevinMicrosoft** [comment]
@vincentche1 from the log sample you sent. It seems like the log header is not compliant with CEF:
![image](https://github.com/Azure/Azure-Sentinel/assets/108271134/316cf9a8-62fc-445c-95e1-de83fa8fef18)

https://www.microfocus.com/documentation/arcsight/arcsight-smartconnectors-8.4/pdfdoc/cef-implementation-standard/cef-implementation-standard.pdf


**Try removing "tmes[1]:" from the log header.**

Also, I think the log date format is also not compliant, but this should not stop the logs from being ingested but might be resulted in inconsistency between the actual log timestamp and CommonSecurityLog timestamp.

**Comment by vincentche1** [comment]
I will check it, thank you for your findings

**Comment by vincentche1** [comment]
Hi Meir, I have checked the sample CEF log message from Microsoft document. We will use 'echo' command to send a mock message to Azure Monitoring Agent (AMA). And the content of the sample log will **not include the date time and host**. 

Link: https://learn.microsoft.com/en-us/azure/sentinel/connect-cef-syslog-ama?tabs=portal#avoid-data-ingestion-duplication
echo -n "<164>**CEF:0|Mock-test|MOCK|common=event-format-test|end|TRAFFIC|1|rt=$common=event-formatted-receive_time**" | nc -u -w0 localhost 514

Hence, the way you suggested above is not applicable. 
In other words, you can consider the content from "CEF..." - which is the real message from device. 
Could you please help us check more and advise? Thank you.

**Comment by MeirLevinMicrosoft** [comment]
In case you don't get even mock messages it's not a formatting issue. I suggest:
1. Make sure you have latest agent version
2. Execute CEF troubleshooter:
_sudo wget -O Sentinel_AMA_troubleshoot.py https://raw.githubusercontent.com/Azure/Azure-sudo wget -O Sentinel_AMA_troubleshoot.py https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Syslog/Sentinel_AMA_troubleshoot.py && sudo python3 Sentinel_AMA_troubleshoot.py&& sudo python3 Sentinel_AMA_troubleshoot.py_

Please note the formatting issue remains IMO, so once you get the mock messages you will still need to fix it.
Also, for further assistance I suggest open a ticket to Azure Sentinel as more data is required to understand the root cause.

**Comment by vincentche1** [comment]
Dear @MeirLevinMicrosoft,
1. The suggestion "remove tmes[1]:" can not be conducted because command send mock message will not contain "tmes[1]".
2. The other CEF logs have been ingested successfully so it is no need to run that command. That command is used when no CEF logs ingests into Sentinel
3. The CEF log is ingested into Syslog table not CommonSecurityLog, please check the image I sent via chat. I cannot attach in comment. Please show me other ways to attach

**Comment by vincentche1** [comment]
Dear @MeirLevinMicrosoft @v-sudkharat , could you let me is there any update or any info you need to check more?

**Comment by vincentche1** [comment]
I am the Sentinel support, please let me know if you need any info

**Comment by v-sudkharat** [comment]
Shared response over Teams. Thanks!
